library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/processor/MutationResolver.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.processor
+
+import com.datadog.android.sessionreplay.model.MobileSegment
+import java.util.LinkedList
+
+internal class MutationResolver {
+
+    fun resolveMutations(
+        prevSnapshot: List<MobileSegment.Wireframe>,
+        currentSnapshot: List<MobileSegment.Wireframe>
+    ): MobileSegment.MobileIncrementalData.MobileMutationData {
+        val prevSnapshotAsMap = prevSnapshot.associateBy { it.id() }.toMutableMap()
+        val updates: MutableList<MobileSegment.WireframeUpdateMutation> = LinkedList()
+        val adds: MutableList<MobileSegment.Add> = LinkedList()
+        prevSnapshot.forEach {
+            prevSnapshotAsMap[it.id()] = it
+        }
+        var prevWireframe: MobileSegment.Wireframe? = null
+        currentSnapshot.forEach { currentWireframe ->
+            val prevMatchedWireframe = prevSnapshotAsMap.remove(currentWireframe.id())
+            if (prevMatchedWireframe != null) {
+                resolveUpdateMutation(prevMatchedWireframe, currentWireframe)?.let {
```

Comment:
It will be null only if the wireframe are matched (in that case it is expected to be null) or if by any chance the wireframe type is not the same for the same view for one snapshot to another. In the latter I have a TODO there to add a log as this behaviour is not expected.